frontend/app/src/store/assets/index.ts

```diff
@@ -163,6 +168,123 @@ export const useAssets = defineStore('assets', () => {
   };
 });
 
+export const useSupportedAssets = defineStore('assets/supported', () => {
+  const supportedAssets = ref<SupportedAsset[]>([]);
+
+  const fetchSupportedAssets = async (refresh: boolean = false) => {
+    set(supportedAssets, []);
+    if (get(supportedAssets).length > 0 && !refresh) {
+      return;
+    }
+    try {
+      const assets = await api.assets.allAssets();
+      set(supportedAssets, convertSupportedAssets(assets));
+    } catch (e: any) {
+      const { notify } = useNotifications();
+      notify({
+        title: i18n
+          .t('actions.balances.supported_assets.error.title')
+          .toString(),
+        message: i18n
+          .t('actions.balances.supported_assets.error.message', {
+            message: e.message
+          })
+          .toString(),
+        display: true
+      });
+    }
+  };
+
+  return {
+    supportedAssets,
+    fetchSupportedAssets
+  };
+});
+
+export const useAssetInfoRetrieval = defineStore(
```

Comment:
Is there any reason why these two stores are not a single one? I would probably have them merged into one.

Reply:
I forgot this one, I was planning to merge them, but at that time I don't have idea for the naming.

Reply:
Okay I think just use `useAssetInfoRetrieval` for them

Reply:
There are several things I need to change, Ill tell u if its ready again.